src/javascripts/modules/pills.js

```diff
@@ -6,28 +6,48 @@ export default class Pills {
   }
 
   setVars() {
-    this.ticking = false
-    this.lastScrollY = 0
+    this.titleBox = document.querySelector('.title-box')
+    this.greenTablet = document.querySelector('.green-tablet')
   }
 
   setupListeners() {
     window.addEventListener('load', () => this.handleLoad())
-    window.addEventListener('scroll', () => this.handleScroll())
+    this.createObserver()
   }
 
   handleLoad() {
     this.timeout = setTimeout(() => this.animatePills(), 500)
     this.releaseSqueezeTimeout = setTimeout(() => this.releasePillSqueeze(), 3500)
   }
 
-  handleScroll() {
-    this.lastScrollY = window.scrollY
-    if (!this.ticking) {
-      window.requestAnimationFrame(() => {
-        this.shiftPills()
-      })
+  createObserver() {
+    this.options = {
+      root: null,
+      rootMargin: "20px",
+      threshold: 0,
     }
-    this.ticking = true
+
+    this.observer = new IntersectionObserver(this.handleIntersect.bind(this), this.options)
+    this.observer.observe(this.greenTablet)
+  }
+
+  handleIntersect(entries, observer) {
+    entries.forEach((entry) => {
+      if (entry.time < 6000) {
+        //preventing pill shift animation onLoad
```

Comment:
This creates an edge case where `entry.time` could be longer than 6000 on first page load if something should for some reason take forever to load. So you might want to have a `this.isPageLoaded` variable that is false by default, and set to true after the page load event. Then you could check that instead of entry.time.

Reply:
With entry.time < 6000, I'm trying to select for just the first intersection observation so that it doesn't trigger a pill shift (intersectionObserver fires onload b/c technically the intersection is happening, although it's not a scroll-to-top situation). If it's allowed to trigger a shift on load, it screws up the loading animation timing. Since entries.length is always 1, I can't tell it to skip the first entry...any idea how to get the same result without something time-based?

Reply:
Agreed, predicting load times seems risky.

Reply:
Since intersection observer fires on page load, maybe set `this.firstIntersection = false` in setVars, then check for it in your `handleIntersect` method, then set it to true at the end of that method?

Reply:
That did the trick, thank you!

Reply:
I ran a couple tests on the pillbar animation and it looks like any difference is miniscule. Here are representative results:

With requestAnimationFrame:
![screen shot 2018-06-22 at 2 27 06 pm](https://user-images.githubusercontent.com/23386632/41793222-6d43717c-7629-11e8-910e-0cdfffb53c94.jpg)

Without:
![image](https://user-images.githubusercontent.com/23386632/41793208-635973dc-7629-11e8-8028-809efa377804.png)